packages/next/src/server/app-render/static/renderers.node.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  Options as RenderToPipeableStreamOptions,
+  ResumeOptions,
+} from 'react-dom/server.node'
+import type { Options as PrerenderToNodeStreamOptions } from 'react-dom/static.node'
+import type { RenderResult, Renderer } from './renderers'
+import { PassThrough } from 'node:stream'
+import type { PostponedState } from 'react-dom/types'
+
+export class ServerRenderer implements Renderer {
+  private readonly renderToPipeableStream = require('react-dom/server.node')
+    .renderToPipeableStream as typeof import('react-dom/server.node')['renderToPipeableStream']
+
+  constructor(private readonly options: RenderToPipeableStreamOptions) {}
+
+  public async render(children: JSX.Element): Promise<RenderResult> {
+    const stream = this.renderToPipeableStream(children, this.options)
```

Comment:
Use `onShellReady` to make this work like the web version. 
```suggestion
    const stream = this.renderToPipeableStream(children, { ...this.options, onShellReady })
```